server/src/test/java/io/crate/metadata/cluster/AlterTableClusterStateExecutorTest.java

```diff
@@ -102,4 +103,18 @@ public void testAddExistingMetaDoNotAddEmptyValues() throws IOException {
         // DO NOT WANT empty containers and nulls: "{"_meta":{"indices":{},"partitioned_by":[],"primary_keys":null,"check_constraints":null,"constraints":{}}}"
     }
 
+    @Test
+    public void testAddExistingMetaHandlesRoutingColumn() throws IOException {
+        AlterTableRequest request = mock(AlterTableRequest.class);
+        Map<String, Object> mapping = new HashMap<>();
+        mapping.put("routing", "routing_col");
+        when(request.mappingDeltaAsMap()).thenReturn(Collections.singletonMap("_meta", mapping));
```

Comment:
Could this be done without mocking? `AlterTableRequest` is mostly a data class.

We try to avoid mocks as they can lead to a test to become invalid without anybody noticing. Unlikely to be the case here - but for example if we extended `mappingDeltaAsMap` to run additional logic, this test case would never include that logic and may test something that in production isn't possible anymore.

Reply:
Thank you, updated